gcd.py

```diff
@@ -1,13 +1,13 @@
 # Wrong gcd find 5 mistakes
 
 def gcd(a, b):
-    assert a <= 0 and b >= 0
+    assert a >= 0 and b >= 0
     while a and b:
         if a > b:
-            a = a / b
+            a = a % b
         else:
-            b = b / a
-    return min(a, b)
+            b = b % a
+    return a + b
```

Comment:
It's okay because finally "a" or "b" will be 0, and that's why it works correct. But for more clear read it's better to use max(a, b). But it's nuances generally you did a great job!)